### DISCUSSION TITLE
Expand Bike Route Options Based on Distance and Slope

### DISCUSSION THREAD

**Discussion by vbarruec** [organicmaps/organicmaps#6090] [Ideas]
I want to open an issue to work on it as my master's thesis project, which will implement the following functionality:

Expand the bicycle route search functionality to allow users to choose from multiple routes that match their preferences for distance and slope.

Would this feature be interesting to add to the app?

**Comment by biodranik** [comment]
Are you speaking about #5382 #3447 #1183 #5009 issues, but only in a bike context? Our router has a generic code that works for any type of routing. Alternate routes is a very important feature needed by many users. It is not easy to implement, and we would be glad if you help, considering that you know C++ well, and know how A* algorithms work.

**Reply by vbarruec** [reply]
It's similar; I can see that in walking and biking routes, the altitude of the route is displayed (not in car routes). The idea is that when you want to plan a route, multiple options are previewed, showing that altitude and distance, so you can choose the itinerary that best suits your physical condition or goals
